### PR TITLE
[vcpkg/msbuild] parellel build

### PR DESCRIPTION
**Pull request by mheyman**
**Describe the pull request**

Defaults to parallel build for msbuild projects. Prior, only projects built in parallel. With this change, source files within the projects can build in parallel. Ports like activemq-cpp use one vcxproj file for the entire port so without this change, it will build single threaded.

- #### Which triplets are supported/not supported? 
N/A

- #### Does your PR follow the maintainer guide
- Yes
- 
- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
 N/A

### CODE REVIEW

**Comment by autoantwort** [comment]
It is maybe possible to respect [VCPKG_MAX_CONCURRENCY](https://github.com/microsoft/vcpkg/blob/master/docs/users/config-environment.md#vcpkg_max_concurrency)

**Comment by JackBoosY** [comment]
> It is maybe possible to respect [VCPKG_MAX_CONCURRENCY](https://github.com/microsoft/vcpkg/blob/master/docs/users/config-environment.md#vcpkg_max_concurrency)?

I think that's the correct way instead of this changes.

**Comment by mheyman** [comment]
A few lines below the change the `/m` switch already existed. That switch uses up to the max number of CPUs to parallelize vcxproj builds within a solution, not .cpp builds within a project. That switch could be changed to `/m:${VCPKG_MAX_CONCURRANCY}` but I'm not sure if the multi-tooltask which the `/p:UseMultiToolTask ` parameter invokes respects that value. I'm looking for the source to inspect it but have yet to find it...

**Comment by mheyman** [comment]
The latest change assumes VCPKG_MAX_CONCURRENCY turns into a CMake variable somehow - I don't know how to use VCPKG_MAX_CONCURRENCY and I know that, in general, passing environment variables though vcpkg requires a bit of complication

**Comment by mheyman** [comment]
So, it looks like the `VCPKG_MAX_CONCURRENCY` environment variable gets turned into the VCPKG_CONCURRENCY CMake variable! (through a call to [get_concurrency()](https://github.com/microsoft/vcpkg-tool/blob/a23237293d0dfc8b9418c36cfc7fb3b98106167f/src/vcpkg/build.cpp#L629).

I *think* these changes fully respect [VCPKG_MAX_CONCURRENCY](https://github.com/microsoft/vcpkg/blob/master/docs/users/config-environment.md#vcpkg_max_concurrency).

**Comment by mheyman** [comment]
Just verified it respects VCPKG_MAX_CONCURRENCY using [Process Explorer](https://docs.microsoft.com/en-us/sysinternals/downloads/process-explorer)!

**Comment by JackBoosY** [comment]
I hope my colleague will review this modification further because I don't have enough knowledge about msbuild.

File: scripts/cmake/vcpkg_install_msbuild.cmake [hunk 1]
```diff
@@ -143,7 +143,10 @@ function(vcpkg_install_msbuild)
         "/p:VcpkgTriplet=${TARGET_TRIPLET}"
         "/p:VcpkgInstalledDir=${_VCPKG_INSTALLED_DIR}"
         "/p:VcpkgManifestInstall=false"
-        "/m"
+        "/p:UseMultiToolTask=true"
```
Comment (BillyONeal) [inline scripts/cmake/vcpkg_install_msbuild.cmake:146]: What effect does this have, if any, for older Visual Studios without `MultiToolTask`?

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks :)

**Pull request merged**